addons/web/static/src/js/views/search_menus.js

```diff
@@ -414,13 +414,15 @@ return Widget.extend({
     },
     get_groupable_fields: function (fields) {
         var self = this,
-            groupable_types = ['many2one', 'char', 'boolean', 'selection', 'date', 'datetime'];
-
-        _.each(fields, function (field, name) {
+           filter_group_field,
+           groupable_types = ['many2one', 'char', 'boolean', 'selection', 'date', 'datetime'];
+        filter_group_field = _.filter(fields, function(field, name) {
             if (field.store && _.contains(groupable_types, field.type)) {
-                self.groupable_fields[name] = field;
+                field.name = name;
+                return field;
             }
         });
+        self.groupable_fields =_.sortBy(filter_group_field,'string');
```

Comment:
`groupable_fields` not store object any more so set correct(array) datatype in init too